lib/std/Thread/Futex.zig

```diff
@@ -21,160 +19,149 @@ const spinLoopHint = std.atomic.spinLoopHint;
 /// - The caller is unblocked by a matching `wake()`.
 /// - The caller is unblocked spuriously by an arbitrary internal signal.
 ///
-/// If `timeout` is provided, and the caller is blocked for longer than `timeout` nanoseconds`, `error.TimedOut` is returned.
-///
 /// The checking of `ptr` and `expect`, along with blocking the caller, is done atomically
 /// and totally ordered (sequentially consistent) with respect to other wait()/wake() calls on the same `ptr`.
-pub fn wait(ptr: *const Atomic(u32), expect: u32, timeout: ?u64) error{TimedOut}!void {
-    if (single_threaded) {
-        // check whether the caller should block
-        if (ptr.loadUnchecked() != expect) {
-            return;
-        }
+pub fn wait(ptr: *const Atomic(u32), expect: u32) void {
+    @setCold(true);
 
-        // There are no other threads which could notify the caller on single_threaded.
-        // Therefor a wait() without a timeout would block indefinitely.
-        const timeout_ns = timeout orelse {
-            @panic("deadlock");
-        };
+    Impl.wait(ptr, expect, null) catch unreachable;
```

Comment:
why catch unreachable?

I suggest to switch on the error even if there is only one error in the set, and especially if we end up making it `unreachable`. This will make it clear what error conditions are unreachable, or perhaps it will highlight the fact that the error condition should be ignored rather than unreachable.